dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs

```diff
@@ -95,64 +99,53 @@ public async Task CreateCollectionAsync(string collectionName, CancellationToken
     /// <inheritdoc />
     public Task<bool> DoesCollectionExistAsync(string collectionName, CancellationToken cancel = default)
     {
-        var container = this._database.Client.GetContainer(this._databaseName, collectionName);
-        return Task.FromResult(container != null);
+        // Azure Cosmos DB does not support checking if container exists without attempting to create it. This does not break the interface but it is not ideal.
+        return Task.FromResult(false);
     }
 
     /// <inheritdoc />
     public async Task DeleteCollectionAsync(string collectionName, CancellationToken cancel = default)
     {
         var container = this._database.Client.GetContainer(this._databaseName, collectionName);
-        var response = await container.DeleteContainerAsync(cancellationToken: cancel);
-
-        if (response.StatusCode == HttpStatusCode.OK)
-        {
-            this._log.LogInformation("Collection {0} deleted", collectionName);
-        }
-        else
-        {
-            throw new CosmosException("Unable to delete collection", response.StatusCode, 0, collectionName, 0);
-        }
+        _ = await container.DeleteContainerAsync(cancellationToken: cancel);
```

Comment:
Does this throw on unsuccessful status codes?

Reply:
added try-catch, notable exception is Container Not Found